en/docs/Administer/ProductAdministration/ManagingUsersAndRoles/ManagingUserStores/ConfigurePrimaryUserStore/configuring-a-read-only-ldap-user-store.md

```diff
@@ -19,169 +19,241 @@ Follow the given steps to configure a read-only LDAP/AD as the primary user stor
 
-        ``` html/xml
-                    <Property name="UserSearchBase">ou=system</Property> 
+        ``` 
+        GroupSearchBase="ou=system,CN=Users,DC=wso2,DC=test"
         ```
 
-    4.  Set the attribute to use as the username, typically either `cn` or `uid` for LDAP. Ideally, `<Property name="UserNameAttribute">` and `<Property name="UserNameSearchFilter">` should refer to the same attribute. If you are not sure what attribute is available in your user store, check with your LDAP/Active Directory administrator.
+    -   Set the GroupSearchFilter and GroupNameAttributes. For example:
 
-        For example:
-
-        ``` html/xml
-                    <Property name="UserNameAttribute">uid</Property>
+        ``` 
+        GroupSearchFilter="(objectClass=groupOfNames)"
+        GroupNameAttribute="cn"
         ```
 
-    5.  Set the `ReadGroups` property to 'true', if it should be allowed to read roles from this user store. When this property is 'true', you must also specify values for the `GroupSearchBase` , `GroupSearchFilter` and `GroupNameAttribute` properties. If the `ReadGroups` property is set to 'false', only Users can be read from the user store. You can set the configuration to read roles from the user store by reading the user/role mapping based on a membership (user list) or backlink attribute as shown below.
-        To read the user/role mapping based on a membership (This is used by the `ApacheDirectory` server and `OpenLDAP)` :
-
-        -   Enable the `ReadGroups` property.
+    -   Set the `MembershipAttribute` property as shown below:
 
-            ``` html/xml
-                            <Property name="ReadGroups">true</Property>
-            ```
-
-        -   Set the `GroupSearchBase` property to the directory name where the Roles are stored. That is, the roles you create using the management console of your product will be stored in this directory location. Also, when LDAP searches for users, it will start from this location of the directory. For example:
-
-            ``` html/xml
-                            <Property name="GroupSearchBase">ou=system,CN=Users,DC=wso2,DC=test</Property>
-            ```
-
-        -   Set the GroupSearchFilter and GroupNameAttributes. For example:
-
-            ``` html/xml
-                            <Property name="GroupSearchFilter">(objectClass=groupOfNames)</Property>
-                            <Property name="GroupNameAttribute">cn</Property>
-            ```
-
-        -   Set the `MembershipAttribute` property as shown below:
-
-            ``` html/xml
-                            <Property name="MembershipAttribute">member</Property> 
-            ```
+        ``` 
+        MembershipAttribute="member"
+        ```
 
-        To read roles based on a backlink attribute, use thefollowingcodesnipetinsteadofthe above:
+        To read roles based on a backlink attribute, use the following code snipet instead of the above:
```

Comment:
Spelling mistake.
It should be "snippet"

Reply:
Fixed with 491587f